pkg/generate/app/cmd/newapp.go

```diff
@@ -498,6 +498,8 @@ func (c *AppConfig) inferBuildTypes(components app.ComponentReferences) (app.Com
 		case input.ResolvedMatch.Score != 0.0:
 			errs = append(errs, fmt.Errorf("component %q had only a partial match of %q - if this is the value you want to use, specify it explicitly", input.From, input.ResolvedMatch.Name))
 			continue
+		case input.ResolvedMatch.Image != nil && (len(input.ResolvedMatch.Image.Config.User) == 0 || input.ResolvedMatch.Image.Config.User == "root" || input.ResolvedMatch.Image.Config.User == "0"):
+			glog.Warningf("Image %q runs as the 'root' user which may not be permitted by your OpenShift cluster configuration", input.ResolvedMatch.Name)
```

Comment:
Can't say OpenShift, this could be in AEP.  "may be restricted by your cluster administrator".  I think we might want to mention SCCs somehow - hard to say for sure.